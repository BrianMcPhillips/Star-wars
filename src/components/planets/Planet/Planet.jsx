import React from 'react';
import PropTypes from 'prop-types';

const Planet = ({ name, terrain, url }) => (
  <>
    <h1>{name}</h1>
    <h4>{terrain}</h4>
  </>
);

Planet.propTypes = {
  name: PropTypes.string.isRequired,
  terrain: PropTypes.string.isRequired
};

export default Planet;
