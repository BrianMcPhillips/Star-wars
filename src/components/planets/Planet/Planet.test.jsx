import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Planet from './Planet';


describe('Planet component', () => {
  afterEach(() => cleanup());
  it('renders Planet', () => {
    const { asFragment } = render(
      <Planet 
        name="Tatooine"
        terrain="desert"
        url="https://swapi.dev/api/planet/1/"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});