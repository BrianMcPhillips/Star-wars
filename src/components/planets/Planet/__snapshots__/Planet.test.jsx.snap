// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet component renders Planet 1`] = `
<DocumentFragment>
  <h1>
    Tatooine
  </h1>
  <h4>
    desert
  </h4>
</DocumentFragment>
`;
